liblangutil/SourceLocation.h

```diff
@@ -49,6 +51,14 @@ struct SourceLocation
 
 	bool isEmpty() const { return start == -1 && end == -1; }
 
+	std::string text() const
+	{
+		struct SourceLocationError: virtual dev::Exception {};
```

Comment:
What use is an exception class that is not visible?

Reply:
I was actually thinking whether we actually need a new one. Wouldn't `dev::Exception` suffice here?

Reply:
I think it is always good to be more specific.